gradio/blocks.py

```diff
@@ -278,6 +278,14 @@ def get_block_class(self) -> str:
     def get_expected_parent(self) -> type[BlockContext] | None:
         return None
 
+    def breaks_grouping(self) -> bool:
```

Comment:
There are other ways grouping can broken too, e.g. a component with visible=False. In the future we can handle those cases as well, though for that purpose maybe 'is_visible' is a better name (and Timer and State are never visible since they have no HTML).
No strong opinion tho, I think it's fine as is and we can revisit later

Reply:
Ah okay. Yeah would be good to solve properly at some point.